app/deed/server.py

```diff
@@ -99,4 +107,40 @@ def sign_deed(deed_, signature_):
         else:
             abort(status.HTTP_403_FORBIDDEN)
 
-        return jsonify(signature=signature), status.HTTP_200_OK
+        return {'signature': signature}, status.HTTP_200_OK
+
+    @blueprint.route('/deed/<deed_id>/completion', methods=['POST'])
+    def confirm_completion(deed_id):
+        def find_deed():
+            deed_ = Deed.get(deed_id)
+            if deed_ is None:
+                abort(status.HTTP_404_NOT_FOUND)
+            return deed_
+
+        def update_case_status():
+            response = case_api.update_status(deed.id, "Completion confirmed")
+            if response.status_code != status.HTTP_200_OK:
+                abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        def update_deed():
+            try:
+                deed_json = copy.deepcopy(deed.json_doc)
```

Comment:
from @dawidskainos in person:
SQLAlchemy seems to be doing a "smart" thing by only commiting changes for things that have "changed" so to force SQLAlchemy to commit the json_docs changes you need to change it by reference. Thus the deepcopy, which creates a new object and causes a by reference change rather than a by value.


I've proposed putting this functionality in to the Deed class since I don't want users of the Deed class to have to remember this problem, it should be handled for them by Deed class.